app/views/main_content_panel.py

```diff
@@ -1681,50 +1681,54 @@ def _do_upload_list_rentry(self) -> None:
 
     @Slot()
     def _on_do_upload_rimsort_log(self) -> None:
-        ret = upload_data_to_0x0_st(str(AppInfo().user_log_folder / "RimSort.log"))
-        if ret:
-            copy_to_clipboard_safely(ret)
-            dialogue.show_information(
-                title="Uploaded file",
-                text="Uploaded RimSort log to http://0x0.st/",
-                information=f"The URL has been copied to your clipboard:\n\n{ret}",
-            )
-            webbrowser.open(ret)
+        self._upload_log(AppInfo().user_log_folder / "RimSort.log")
 
     @Slot()
     def _on_do_upload_rimsort_old_log(self) -> None:
-        ret = upload_data_to_0x0_st(str(AppInfo().user_log_folder / "RimSort.old.log"))
-        if ret:
+        self._upload_log(AppInfo().user_log_folder / "RimSort.old.log")
+
+    @Slot()
+    def _on_do_upload_rimworld_log(self) -> None:
+        player_log_path = (
+            Path(
+                self.settings_controller.settings.instances[
+                    self.settings_controller.settings.current_instance
+                ].config_folder
+            ).parent
+            / "Player.log"
+        )
+
+        self._upload_log(player_log_path)
+
+    def _upload_log(self, path: Path) -> None:
+        if not os.path.exists(path):
+            dialogue.show_warning(
+                title="File not found",
+                text="The file you are trying to upload does not exist.",
+                information=f"File: {path}",
+            )
+            return
+
+        success, ret = self.do_threaded_loading_animation(
+            gif_path=str(AppInfo().theme_data_folder / "default-icons" / "rimsort.gif"),
+            target=partial(upload_data_to_0x0_st, str(path)),
+            text="Uploading to 0x0.st...",
+        )
+
+        if success:
             copy_to_clipboard_safely(ret)
             dialogue.show_information(
                 title="Uploaded file",
-                text="Uploaded RimSort log to http://0x0.st/",
+                text="Uploaded log to http://0x0.st/",
```

Comment:
change to 
text=f"Uploaded {path} to http://0x0.st/",

Reply:
Doing path.name since the path itself has the potential to be very long.